keps/sig-network/20191205-kube-proxy-IP-node-binding.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Give the ability to the LoadBalancer provisioner or a user to disable the LB IP binding
+authors:
+  - "@Sh4d1"
+owning-sig: sig-network
+participating-sigs:
+  - sig-cloud-provider
+reviewers:
+  - "@thockin"
+  - "@cheftako"
+  - TBD
+approvers:
+  - "@thockin"
+  - TBD
+editor: TBD
+creation-date: 2019-12-05
+last-updated: 2019-12-21
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Give the ability to the LoadBalancer provisioner or a user to disable the LB IP binding
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+<!-- /toc -->
+
+## Summary
+
+The different kube-proxy implementations (at least ipvs and iptables), as of today, are binding the External IPs of the LoadBalancer Service to each node. (iptables are creating some rules to redirect packets directly to the service and ipvs is binding the IP to one interface on the node). This feature exists because:
+
+1. To redirect pod traffic that is intended for LoadBalancer IP directly to the service, bypassing the loadbalancer for path optimization.
+2. Some LBs are sending packet with the destination IP equal to the LB IP, so the IP needs to be routed directly to the right service (otherwise looping back and forth)
+3. (TBD)
+
+This enhancement proposes a way to make these features configurable, giving the cloud provider, or the user, a way to disable the behaviour of kube-proxy.
+The best way would be to set a default behaviour at the cloud controller level, that can be overwritten by the user at a service level.
+
+## Motivation
+
+There are numerous problems with the current behaviour:
+
+1. Some cloud providers (Scaleway, ...) are sometimes using the LB's external IP as source IP when sending packets to the cluster. This is a problem in the ipvs mode of kube-proxy since the IP is bounded to an interface and healthchecks from the LB are never coming back.
```

Comment:
Also Tencent cloud based on some convos here https://github.com/kubernetes/kubernetes/issues/79783, cc @imroc

Reply:
Yes, Tencent Cloud also uses LB's external IP as source IP in healthchecking

Reply:
I'll add Tencent Cloud to the list then 😄